docs/cddl/diffs/common/input-blocks.md

```diff
@@ -0,0 +1,58 @@
+# Input Blocks - Leios CDDL
+
+**Single IB Limit**: Each producer can generate **at most one Input Block per slot**, following the crypto-benchmarks implementation approach rather than the full Poisson sortition used in simulations.
+
+**VRF Lottery**: Eligibility uses the simplified probability model:
+
+$$P = 1 - e^{-f_{IB} \cdot \sigma}$$
+
+Where $f_{IB}$ is the per-slot IB generation rate and $\sigma$ is the producer's stake fraction.
+
+<sub>[1] [Crypto-benchmarks Sortition](https://github.com/input-output-hk/ouroboros-leios/blob/main/crypto-benchmarks.rs/Specification.md?plain=1#L64), [2] [Rust Simulation IB Generation](https://github.com/input-output-hk/ouroboros-leios/blob/main/sim-rs/sim-core/src/sim/node.rs#L561-L597)</sub>
+
+## Block Structure
+
+```cddl
+ input_block =
+   [ ib_header                  : ib_header
+   , transaction_bodies         : [* transaction_body]
+   , transaction_witness_sets   : [* transaction_witness_set]
+   , auxiliary_data_set         : {* transaction_index => auxiliary_data}
+   , invalid_transactions       : [* transaction_index]
+   ]
+```
+<sub>[1] [Formal Spec](https://github.com/input-output-hk/ouroboros-leios-formal-spec/blob/main/formal-spec/Leios/Blocks.agda#L40-L57), [2] [Haskell Simulation](https://github.com/input-output-hk/ouroboros-leios/blob/main/simulation/src/LeiosProtocol/Common.hs#L138-L142), [3] [Rust Simulation](https://github.com/input-output-hk/ouroboros-leios/blob/main/sim-rs/sim-core/src/model.rs#L136-L141)</sub>
+
+## Header Structure
+
+```cddl
+ ib_header =
+   [ ib_header_body       : ib_header_body
+   , body_signature       : kes_signature
+   ]
+ 
+ ib_header_body =
+   [ slot                 : slot_no                                  ; Slot when IB was created
+   , producer_id          : pool_id                                  ; Block producer identifier
+   , vrf_proof            : vrf_cert                                 ; VRF proof of eligibility to produce IB
+   , block_body_hash      : hash32                                   ; Hash of the block body
+   , ? ranking_block_ref  : hash32                                   ; Reference to ranking block for ledger state
```

Comment:
Why is this optional? And have we definitely decided that IBs reference RBs. (We had a lot of discussion in Edinburgh about IBs referencing EBs etc.) I do think that IBs should reference RBs, and that should not be optional, but perhaps we haven't sufficiently thought through the implications of forks.